cli/src/pcluster/resources/custom_resources/OpenSearch_lambda/Lambda_pushing_events_to_OpenSearch.py

```diff
@@ -43,6 +44,23 @@ def transform(payload):
         index_name = "cwl-" + ".".join(index_name)
 
         source = build_source(log_event["message"], log_event.get("extractedFields", None))
+        if source["event-type"] == "scontrol-show-job-information":
+            query = {
+                "size": 1,
+                "query": {"match": {"event-type.keyword": "node-instance-mapping-event"}},
+                "sort": [{"datetime": {"order": "desc"}}],
+            }
+            response = json.loads(get(query).text)
+            node_name = source["detail"]["node_list"]
+            node_name_list = get_node_list(node_name)
+            node_map = response["hits"]["hits"][0]["_source"]["detail"]["node_list"]
+            nodes = []
+            node_dict = get_node_dict(node_map)
+            for node in node_name_list:
+                nodes.append(node_dict[node])
+            source["detail"]["nodes"] = nodes
+            del source["detail"]["node_list"]
```

Comment:
We should move this section to its own function, what do you think?

Reply:
Done